### PR TITLE
Fixups to allow multiple TUF root signatures

### PR DESCRIPTION
**Pull request by vkhoroz**
Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by vkhoroz** [comment]
@doanac @mike-sul I found these two things when testing my new code allowing to add more TUF keys.

File: subcommands/keys/tuf_utils.go [hunk 1]
```diff
@@ -361,15 +363,27 @@ func genProdTufRoot(ciRoot *client.AtsTufRoot) (prodRoot *client.AtsTufRoot) {
...
-	// Always sign with new root key; sign with old root key if it was rotated.
-	oldKey, err := FindOneTufSigner(curCiRoot, creds, curCiRoot.Signed.Roles["root"].KeyIDs)
-	subcommands.DieNotNil(err)
-	newKey, err := FindOneTufSigner(newCiRoot, creds, newCiRoot.Signed.Roles["root"].KeyIDs)
-	subcommands.DieNotNil(err)
-	signers := []TufSigner{newKey}
-	if oldKey.Id != newKey.Id {
-		signers = append(signers, oldKey)
+	// Find new and old keys that match; at least one of them must be found.
+	oldKey, oldErr := FindOneTufSigner(curCiRoot, creds, curCiRoot.Signed.Roles["root"].KeyIDs)
```
Comment (mike-sul) [inline subcommands/keys/tuf_utils.go:379]: If `oldErr` is not `errFoundNoKey` does it make sense to continue?

File: subcommands/keys/tuf_utils.go [hunk 2]
```diff
@@ -361,15 +363,27 @@ func genProdTufRoot(ciRoot *client.AtsTufRoot) (prodRoot *client.AtsTufRoot) {
...
+	newKey, newErr := FindOneTufSigner(newCiRoot, creds, newCiRoot.Signed.Roles["root"].KeyIDs)
```
Comment (mike-sul) [inline subcommands/keys/tuf_utils.go:380]: if `newErr` is not `errFoundNoKey` we can `subcommands.DieNotNil(newErr)` at this point?

Comment (vkhoroz) [reply subcommands/keys/tuf_utils.go:380]: I played with this code quite a bit.
What I have is the minimal and clear enough variant.

Initially, I was doing it like you suggest, and that ended up into too many "if" statements, making it hard to follow.

Comment (mike-sul) [reply subcommands/keys/tuf_utils.go:380]: ok, that's what I thought.

**Review by mike-sul — APPROVED** [review]

File: subcommands/keys/tuf_utils.go (hunk 2 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_utils.go]: Actually, I tried to reshuffle it in a bit more plausible way: https://github.com/foundriesio/fioctl/pull/240/commits/ff987dce2edb913341220d3a7e617f4cdaeea2fe

Please, let me know if that looks better to you.

**Review by doanac — APPROVED** [review]
